timemachine/potentials/nonbonded.py

```diff
@@ -19,12 +19,24 @@
 
 Array = Any
 
+from typing import Optional
 
-def switch_fn(dij, cutoff):
-    return jnp.power(jnp.cos((jnp.pi * jnp.power(dij, 8)) / (2 * cutoff)), 2)
 
+def switch_fn(dij, cutoff=None):
+    """heuristic switching function
 
-from typing import Optional
+    intended to:
+    * have {f, f', f''} go to 0 at cutoff
+    * keep "switch_fn(dij) * erfc(beta * dij)" as close as possible to "erfc(beta * dij)"
+        for the range dij in [0, 1.2), for beta = 2.0
+
+    not necessarily intended for use with LJ
+    """
+    if cutoff is not None and jnp.isfinite(cutoff):
```

Comment:
This function can't be JIT compiled as written. Would it be acceptable to encode "no cutoff" as `cutoff=np.inf` and remove the branch here?

Reply:
Oof, good call. A late thought was to avoid `dij / None`, but this was broken. In any case, it seems cleanest to do this by encoding "no cutoff" as np.inf, as you suggest. Applied in https://github.com/proteneer/timemachine/pull/1290/commits/6edfa3ff40bc2968cdd760b5882b5e64508647a5